src/scipp/io/hdf5.py

```diff
@@ -215,7 +215,7 @@ class DataArrayIO:
     @staticmethod
     def write(group, data):
         _write_scipp_header(group, 'DataArray')
-        group.attrs['name'] = data.name
+        group.attrs['scipp-name'] = data.name
```

Comment:
Why `scipp-name` instead of just `name`?

Reply:
My initial thought was that if we add support for HDF5-style attributes, we need to avoid name clashes with user-defined attrs. But I have since realised that we should not mix user-attrs with our own. Because this would again limit the names of attrs to ASCII. And also because we cannot guarantee that there are no name clashes.

Should I change it back?

Reply:
Good point on the attributes... I think you are probably right that we should not store attributes of scipp objects (which are not supported yet) as HDF5 attributes? As you say, prefixing things does not fix this. So I would use `name` for clarity.